### PR TITLE
Bug: Made it so that 0 was included in uint8

### LINKED ISSUE #14401: BUG: pd.to_numeric(..., downcast='unsigned') should accept 0
with 0.19.0.

```
In [28]: pd.to_numeric(pd.Series([1,2,3]), downcast='unsigned')
Out[28]: 
0    1
1    2
2    3
dtype: uint8

In [29]: pd.to_numeric(pd.Series([0,1,2,3]), downcast='unsigned')
Out[29]: 
0    0
1    1
2    2
3    3
dtype: int64
```

```
In [30]: np.iinfo(np.uint8).min
Out[30]: 0
```

I think 0 is a legit uint8, so [29] should be uint8.

### PR DESCRIPTION
**Pull request by verhalenn**
- [ ] closes #14401 
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

Decided to restart. Sorry for the inconvenience. -> #14472

### CODE REVIEW

**Comment by sinhrks** [comment]
Thx for the PR. pls rebase existing PR from next.

File: doc/source/whatsnew/v0.19.1.txt [hunk 1]
```diff
@@ -45,3 +45,4 @@ Bug Fixes
 
 - Bug in ``pd.concat`` where names of the ``keys`` were not propagated to the resulting ``MultiIndex`` (:issue:`14252`)
 - Bug in ``MultiIndex.set_levels`` where illegal level values were still set after raising an error (:issue:`13754`)
+- Bug in ``pd.to_numeric`` where it would not downcast a 0 to a uint8 (:issue:`14404`)
```
Comment (sinhrks) [inline doc/source/whatsnew/v0.19.1.txt]: pls include #14401.

File: pandas/tools/tests/test_util.py [hunk 2]
```diff
@@ -391,6 +391,14 @@ def test_downcast(self):
             res = pd.to_numeric(data, downcast=downcast)
             tm.assert_numpy_array_equal(res, expected)
 
+        #check that 0 works as a unsigned downcast
+
+        data = [0, 1, 2, 3]
```
Comment (jreback) [inline pandas/tools/tests/test_util.py]: check this for all of the downcast parameters (where it *should* be included)

Comment (verhalenn) [inline pandas/tools/tests/test_util.py]: Did you want me to just edit everything between line 327 and line 368 in that test to include 0's in the arrays? Trying it before and after, it fails before and passes after. That being said if you don't want me to touch it for some reason it's relatively easy to make a new one. Also is there a reason that 340 and 341 is redundant?

**Comment by verhalenn** [comment]
Added a series of tests which applied a pandas series that included the minimum and maximum values of each dtype to a to_numeric downcast to see if it would downcast to the appropriate dtype. It did not add to uint64 but I'm not sure if there were plans to include uint64 so I'll open up a new issue. Also edited the whatsnew.

File: doc/source/whatsnew/v0.19.1.txt [hunk 3]
```diff
@@ -45,3 +45,4 @@ Bug Fixes
 
 - Bug in ``pd.concat`` where names of the ``keys`` were not propagated to the resulting ``MultiIndex`` (:issue:`14252`)
 - Bug in ``MultiIndex.set_levels`` where illegal level values were still set after raising an error (:issue:`13754`)
+- Bug in ``pd.to_numeric`` where it would not downcast a 0 properly. (:issue:`14401`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.1.txt]: say ``downcast='unsigned'`` as a kwargs is passed

File: pandas/tools/tests/test_util.py [hunk 4]
```diff
@@ -401,6 +401,32 @@ def test_downcast(self):
             res = pd.to_numeric(data, downcast=downcast)
             tm.assert_numpy_array_equal(res, expected)
 
+        # check that the smallest and largest values in each integer type pass to each type.
+        int8 = [-128, 127]
```
Comment (jreback) [inline pandas/tools/tests/test_util.py]: don't do this explicty, use ``np.iinfo(dtype).min/max`` as these vary by platform

Comment (jreback) [inline pandas/tools/tests/test_util.py]: actually on 2nd though, might be ok if appveyor passes

Comment (jreback) [inline pandas/tools/tests/test_util.py]: can you do this in a loop instead (eg. a loop of tuples of the start,stop, dtype)

Comment (verhalenn) [inline pandas/tools/tests/test_util.py]: I was actually thinking dictionary if that's okay. And I believe that the numpy types are standard across platforms but I'm not positive which is why I didn't bother with the np.info(dtype).min/max but for safeties sake I'll include it.

File: doc/source/whatsnew/v0.19.1.txt [hunk 5]
```diff
@@ -48,3 +48,6 @@ Bug Fixes
 - Bug in ``MultiIndex.set_levels`` where illegal level values were still set after raising an error (:issue:`13754`)
 - Bug in ``DataFrame.to_json`` where ``lines=True`` and a value contained a ``}`` character (:issue:`14391`)
 - Bug in ``df.groupby`` causing an ``AttributeError`` when grouping a single index frame by a column and the index level (:issue`14327`)
+- Bug in ``pd.to_numeric`` where it would not downcast a 0 to a uint8 (:issue:`14404`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.1.txt]: just make a single entry

File: pandas/tools/tests/test_util.py [hunk 6]
```diff
@@ -401,6 +401,62 @@ def test_downcast(self):
             res = pd.to_numeric(data, downcast=downcast)
             tm.assert_numpy_array_equal(res, expected)
 
+        # check that the smallest and largest values in each integer type pass to each type.
```
Comment (jreback) [inline pandas/tools/tests/test_util.py]: this is still overly verbose. do something like.

```
checks = [('int8', 'integer', [np.iinfo(np.int8).min, np.iinfo(np.int8).max]), 
                  .....
               ]

for dtype, downcast, min_max in checks:
     ....
```

**Pull request closed without merging**